sharkfuser/include/fusilli/graph/graph.h

```diff
@@ -103,8 +103,75 @@ class Graph : public INode {
   // Executes the graph using IREE runtime. Requires a `variantPack` which is a
   // map from `TensorAttr` to `Buffer` wrapping the `iree_hal_buffer_view_t *`.
   // Definition in `fusilli/backend/runtime.h`.
+  //
+  // Backend Specific Execution Behavior
+  //   For some backends execution will be async. The specifics of how one
+  //   should launch a kernel and synchronize work items vary per backend.
+  //
+  // CPU
+  //   Synchronous execution, no synchronization necessary.
+  //
+  // AMDGPU
+  //   Asynchronous execution. Synchronization will depend on if you're using an
+  //   external hip stream.
+
+  //   `Handle::create(Backend::AMDGPU)`
+  //      When not using an external hip stream, kernel launches are async.
+  //      Reads, writes, and allocations done through the fusilli APIs
+  //      (`fusilli::Buffer::allocate`, `fusilli::Buffer::read`, etc.) are
+  //      stream ordered (aka happen in the order executed). Read implicitly
+  //      synchronizes (waits for anything in the stream behind it to finish)
+  //      ensuring correct data is read. Any GPU read or write done outside of
+  //      the fusilli APIs would lead to undefined behavior.
```

Comment:
Probably worth highlighting the main difference (I think) is that this uses the default (null) stream which implicitly synchronizes against all other streams (which is still OK if no other stream is used in the same process), whereas the case below uses an external stream and doesn't suffer from the same restriction.